ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -11,14 +11,8 @@
 //!     partial register stalls in the CPU pipeline. Generally speaking we prefer to use 64-bit
 //!     operations, but sometimes we special-case 32-bit operations since they are so common.
 //!
-//!   * If an object occupies only part of a 64-bit register, then there are no guarantees about
-//!     the unused higher-order bits. Codegen routines must be mindful of this and, depending upon
-//!     the operation, may be required to (for example) "mask off" or "sign extend into" the
-//!     undefined bits in order to compute the correct result.
-//!
-//! FIXME: the code generator clobbers registers willy-nilly because at the time of writing we have
-//! a register allocator that doesn't actually use any registers. Later we will have to audit the
-//! backend and insert register save/restore for clobbered registers.
+//!   * When a value is in a register, we make no guarantees about what the upper bits are set to.
```

Comment:
I know what you mean here, but we should be clear that the upper undefined bits are *unused* by the value, and that the value is encoded in the lower order bits.

Reply:
I wonder if we should also give an example of a typical scenario where zero/sign extend is necessary. That would help us check we are on the same page and help newcomers to understand what we mean.

My understaning is that:
 - inputs to operations (where the result would be incorrect if we didn't) must define the undefined bits with sign/zero extend (or equivalent).
 - outputs of operations may leave the undefined bits undefined: they may assume that any operation consuming the value will define the upper bits appropriately for the operation in question.

Reply:
The phrasing I'm using is a fairly standard phrasing from specifications. I would prefer to follow that lead rather than introduce our own phrasing that people will not be familiar with.

Reply:
OK.  I would have liked a bit more explanation, but i suppose we have bigger fish to fry.